index.js

```diff
@@ -7,4 +13,62 @@ const fgMultipleHooks = async (req, res, ...multipleHooks) => {
     return shouldAbort;
 };
 
-module.exports = fgMultipleHooks;
\ No newline at end of file
+const defaultLastOnResponse = async (req, res, stream) => {
+    const chunked = stream.headers[TRANSFER_ENCODING_HEADER_NAME]
+        ? stream.headers[TRANSFER_ENCODING_HEADER_NAME].endsWith('chunked')
+        : false
+
+    if (req.headers.connection === 'close' && chunked) {
+        try {
+            // remove transfer-encoding header
+            const transferEncoding = stream.headers[TRANSFER_ENCODING_HEADER_NAME].replace(/(,( )?)?chunked/, '')
+            if (transferEncoding) {
+                // header format includes many encodings, example: gzip, chunked
+                res.setHeader(TRANSFER_ENCODING_HEADER_NAME, transferEncoding)
+            } else {
+                res.removeHeader(TRANSFER_ENCODING_HEADER_NAME)
+            }
+
+            if (!stream.headers['content-length']) {
```

Comment:
what about only this block as the implementation for this hook?

Reply:
I would like the consumer to have an option to create a hook with no stream responsibility. having the whole functionality in a default setup allows me to do that

Reply:
have implemented a different style, i think it covers this flexibility